R/plotFeatureLoadings.R

```diff
@@ -0,0 +1,404 @@
+#' tse <- runPCA(tse, name = "PCA", ncomponents = 2)
+#' plotFeatureLoadings(tse, method= "PCA", plot = "Tree")
+#' 
+#' 
+#' # Plotting without tree as a heatmap
+#' library(mia)
+#' library(ggtree)
+#' library(scater)
+#' data("GlobalPatterns", package = "mia")
+#' tse <- GlobalPatterns
+#' tse <- logNormCounts(tse)
+#' tse <- agglomerateByPrevalence(tse, rank="Phylum", prevalence=0.99, update.tree = TRUE)
+#' tse <- runPCA(tse, name = "PCA", ncomponents = 2)
+#' plotFeatureLoadings(loadings_matrix, method= "PCA", plot = "Heatmap")
+#' 
+#' # Plotting without tree as a barplot
+#' library(mia)
+#' library(ggtree)
+#' library(scater)
+#' data("GlobalPatterns", package = "mia")
+#' tse <- GlobalPatterns
+#' tse <- logNormCounts(tse)
+#' tse <- agglomerateByPrevalence(tse, rank="Phylum", prevalence=0.99, update.tree = TRUE)
+#' tse <- runPCA(tse, name = "PCA", ncomponents = 2)
+#' plotFeatureLoadings(loadings_matrix, method= "PCA", plot = "Barplot")
+#' 
+#' #' # Plotting without tree as a screeplot
+#' library(mia)
+#' library(ggtree)
+#' library(scater)
+#' data("GlobalPatterns", package = "mia")
+#' tse <- GlobalPatterns
+#' tse <- logNormCounts(tse)
+#' tse <- agglomerateByPrevalence(tse, rank="Phylum", prevalence=0.99, update.tree = TRUE)
+#' tse <- runPCA(tse, name = "PCA", ncomponents = 2)
+#' plotFeatureLoadings(loadings_matrix, method= "PCA", plot = "Screeplot")
+NULL
+
+#' @rdname plotFeatureLoadings
+setGeneric("plotFeatureLoadings", signature = c("object"),
+           function(object, ...)
+             standardGeneric("plotFeatureLoadings"))
+
+
+#' @rdname plotFeatureLoadings
+#' @export 
+setMethod("plotFeatureLoadings", signature = c(object = "TreeSummarizedExperiment"),
+          function(object,
+                   method = "PCA", plot = "Heatmap",
+                   ...){
+            
+            .check_parameters(object,
+                              method = method, plot = plot, ...)   
+            
+            if (method == "PCA") {
+              loadings_matrix <- attr(reducedDim(tse, "PCA"), "rotation")
+              df <- data.frame(loadings_matrix)
+              df[["Feature"]] <- rownames(loadings_matrix)
+              df[["Feature"]] <- factor(df[["Feature"]], levels = df[["Feature"]])
+              df <- df[ order(abs(df[["PC1"]]), decreasing = TRUE)[1:10], ]
+              df <- df[ order(df[["PC1"]]), ]
+              df <- df[ order(abs(df[["PC2"]]), decreasing = TRUE)[1:10], ]
+              df <- df[ order(df[["PC2"]]), ]
+            } else {
+              stop("These methods are yet to be implemented", call. = FALSE)
+            }
+            
+            if (plot == "Tree") {
+              
+              phylo <- rowTree(object)
+              circ <- ggtree(phylo, layout = "circular")
+              df <- rowData(object)
+              color <- randomcoloR::distinctColorPalette(
+                length(
+                  unique(
```

Comment:
Try to restructure the code. The "main" function should be simple and call internal functions. Here:

1. You could first call `.get_loadings_plot_data` that prepares data.frame from the data
2. The you could call `.loadings_plotter` That creates the plot.